src/main/java/org/osc/manager/rest/server/api/SecurityApis.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    @Path("/{sgId}")
+    @DELETE
+    public void deleteSecurityGroup(@PathParam("sgId") Long sgId) throws Exception {
+        LOG.info(String.format("Deleting the security group for id %s ", "" + Long.toString(sgId)));
+        this.txControl.required(new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                SecurityApis.this.sgApi.deleteSecurityGroup(Long.toString(sgId));
+                return null;
+            }
+        });
+    }
+
+    @GET
+    public List<String> getSecurityGroupIds() throws Exception {
+        LOG.info("Listing security group ids'");
+        List<? extends ManagerSecurityGroupElement> securityGroups = this.sgApi.getSecurityGroupList();
+        List<String> sgList = new ArrayList<String>();
+        if (securityGroups.isEmpty()) {
+            return sgList;
+        }
+        sgList = securityGroups.stream().map(ManagerSecurityGroupElement::getSGId).collect(Collectors.toList());
+        return sgList;
+    }
+
+    @Path("/{sgId}")
+    @GET
+    public SecurityGroupEntity getSecurityGroup(@PathParam("sgId") Long sgId) throws Exception {
+        LOG.info(String.format("Getting the security group for id %s ", "" + Long.toString(sgId)));
+        return this.sgApi.getSecurityGroup(Long.toString(sgId));
+    }
+
+    @Path("/{sgId}/sgIntf")
+    @POST
+    public String createSecurityGroupInterface(@PathParam("sgId") Long sgId, SecurityGroupInterfaceEntity entity)
+            throws Exception {
+        LOG.info(String.format("Creating the security group interface with name %s ", "" + entity.getName()));
+        SecurityGroupEntity sgElement = new SecurityGroupEntity();
+        sgElement.setId(sgId);
+        entity.setSecurityGroup(sgElement);
+        return this.sgiApi.createSecurityGroupInterface(entity);
+    }
+
+    @Path("/{sgId}/sgIntf/{sgIntfId}")
+    @PUT
+    public SecurityGroupInterfaceElement updateSecurityGroupInterface(@PathParam("sgId") Long sgId,
+            @PathParam("sgIntfId") Long sgIntfId, SecurityGroupInterfaceEntity entity) throws Exception {
+        LOG.info(String.format("Updating the security group interface with sgid %s ; sginterfaceid %s",
+                "" + Long.toString(sgId), "" + Long.toString(sgIntfId)));
+        SecurityGroupEntity sgElement = new SecurityGroupEntity();
+        sgElement.setId(sgId);
+        entity.setSecurityGroup(sgElement);
+        entity.setId(sgIntfId);
+        this.sgiApi.updateSecurityGroupInterface(entity);
+        return entity;
+    }
+
+    @Path("/{sgId}/sgIntf/{sgIntfId}")
+    @DELETE
+    public void deleteSecurityGroupInterface(@PathParam("sgId") Long sgId, @PathParam("sgIntfId") Long sgIntfId)
+            throws Exception {
+        LOG.info(String.format("Deleting the security group interface with sgid %s ; sginterfaceid %s",
+                "" + Long.toString(sgId), "" + Long.toString(sgIntfId)));
+        this.txControl.required(new Callable<Void>() {
```

Comment:
The transaction should not be needed here either.

Reply:
I was getting 500 error response and I believe this was happening that rest was sending the response before delete transaction was completed. This is due to plugin implementation having two independent transaction, first one for query and second for delete request.
OK, I will fix this at the plugin level.

Reply:
remark as above.